bigframes/series.py

```diff
@@ -680,6 +680,17 @@ def rdivmod(self, other) -> Tuple[Series, Series]:  # type: ignore
         return (self.rfloordiv(other), self.rmod(other))
 
     def __matmul__(self, other):
+        if isinstance(other, bigframes.dataframe.DataFrame):
+            return Series(
+                [
+                    pandas.NA if other[col].isna().any() else (self * other[col]).sum()
+                    for col in other.columns
+                ],
+                index=other.columns,
+                name=self.name,
+            )
```

Comment:
This will perform several queries, as each aggregate `.any()` operation initiates a query. I think we can do this more efficiently.

Reply:
Send an optimization, PTAL. It doesn't apply to multi-index yet due to b/313747368, left a TODO.